crates/lib/src/install.rs

```diff
@@ -1653,22 +1657,26 @@ struct RootMountInfo {
 
 /// Discover how to mount the root filesystem, using existing kernel arguments and information
 /// about the root mount.
-fn find_root_args_to_inherit(cmdline: &Cmdline, root_info: &Filesystem) -> Result<RootMountInfo> {
-    let root = cmdline
-        .value_of_utf8("root")
-        .context("Parsing root= karg")?;
+fn find_root_args_to_inherit(
+    cmdline: &bytes::Cmdline,
+    root_info: &Filesystem,
+) -> Result<RootMountInfo> {
     // If we have a root= karg, then use that
+    let root = cmdline
+        .find("root")
+        .map(utf8::Parameter::try_from)
```

Comment:
How about `find_utf8` which handles this since that's going to be the 90% case - us looking for a utf8 key+value?

That would be best implemented in terms of `iter_utf8` on the bytes one which would basically be `.iter().filter(utf8::Parameter::try_from.ok())` right?

Reply:
Hm I implemented this, but now we don't return an error in the case where kargs is `root=invalid\xffutf8`, and instead fall back to the case where `root` doesn't appear at all.  Desirable?  I'm not sure.